### PR TITLE
Add variable types plotting, and variable page to docs

### LINKED ISSUE #990: Create UML-Diagram in Docs for Variable Types
### Bug/Feature Request Title
- Create a UML diagram of all Variables, and put it in the documentation
- This diagram would allow the user to see all the VTypes and subtypes (Datetime --> DateofBirth, Categorical --> Ordinal, etc). 
- We can add this to the documentatio via rst.

### PR DESCRIPTION
**Pull request by gsheni**
### Pull Request Description
- Fixes #990
- Created `list_variable_types` function (similar to list_primitives)
- Separates out the EntitySet plotting into separate utils file to be used by other functions
- Moves `find_variable_types` into utils of `variables`

### CODE REVIEW

**Comment by gsheni** [comment]
Variable Types plot:
variables

**Comment by gsheni** [comment]
Page on Docs:
https://docs.featuretools.com/en/add_variable_page/variables.html

File: docs/source/changelog.rst [hunk 1]
```diff
@@ -4,12 +4,15 @@ Changelog
 ---------
 .. **Future Release**
```
Comment (rwedge) [inline docs/source/changelog.rst:5]: uncomment the Future Release section

File: docs/source/variables.ipynb [hunk 2]
```diff
@@ -0,0 +1,182 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Variable Types\n",
+    "\n",
+    "A Variable is analogous to a column in table in a relational database. When running DFS, Featuretools will attempt to infer the types of variable present in your data. DFS also allows for specifying the variable types.\n",
```
Comment (rwedge) [inline docs/source/variables.ipynb:9]: > When running DFS, Featuretools will attempt to infer the types of variable present in your data.

types of _variables_

I think in this context DFS should be replaced with Entity or EntitySet

File: docs/source/variables.ipynb [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+    "\n",
+    "### DateOfBirth\n",
+    "A more distinct type of datetime is a DateOfBirth. This is an important distinction because it allows additional primitives to be applied to the data to generate new features. For example, having an DateOfBirth variable type, will allow the Age primitive to be applied during DFS, and lead to a new Numeric feature."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## TextT\n",
```
Comment (rwedge) [inline docs/source/variables.ipynb:107]: extra T in Text

File: docs/source/variables.ipynb [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+    "Text is a long-form strings, that can be of any length. It is commonly used with NLP operations, such as TF-IDF. Featuretools supports NLP operations with nlp-primitives [add-on](https://innovation.alteryx.com/natural-language-processing-featuretools/). Text variable type can be represented as strings. "
```
Comment (rwedge) [inline docs/source/variables.ipynb:108]: I think the last sentence is repeating information found in the first sentence.

1st and 3rd sentence could use an extra word or two.

File: featuretools/variable_types/utils.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        parents = [y for y in inspect.getmro(x) if y not in [object, x]]
+        if parents == [Variable]:
+            # a direct child of Variable
+            graph.edge(Variable.__name__, x.__name__)
+        else:
+            # a descent of Variable, only plot the parent - child relation
+            graph.edge(parents[0].__name__, x.__name__)
+
+    # NumericTimeIndex and DatetimeTimeIndex are different since they subclassed under 2 variables
+    for x in [NumericTimeIndex, DatetimeTimeIndex]:
```
Comment (rwedge) [inline featuretools/variable_types/utils.py:90]: Could we handle this with less hard-coding so if we add more Variables with multiple subclasses in the future the logic will still work?

File: docs/source/variables.ipynb (hunk 3 above)
Comment (gsheni) [reply docs/source/variables.ipynb:107]: Fixed

File: docs/source/changelog.rst (hunk 1 above)
Comment (gsheni) [reply docs/source/changelog.rst:5]: Fixed

File: docs/source/variables.ipynb [hunk 6]
```diff
@@ -6,7 +6,7 @@
    "source": [
     "# Variable Types\n",
     "\n",
-    "A Variable is analogous to a column in table in a relational database. When running DFS, Featuretools will attempt to infer the types of variable present in your data. DFS also allows for specifying the variable types.\n",
+    "A Variable is analogous to a column in a table in a relational database. When running DFS, Featuretools will attempt to infer the types of variables present in each Entity of the EntitySet. DFS also allows for specifying the variable types.\n",
```
Comment (rwedge) [inline docs/source/variables.ipynb:9]: It depends on how technical we want to get here, but inferring variable types happens before running DFS.

File: docs/source/variables.ipynb [hunk 7]
```diff
@@ -104,8 +104,8 @@
    "cell_type": "markdown",
    "metadata": {},
    "source": [
-    "## TextT\n",
-    "Text is a long-form strings, that can be of any length. It is commonly used with NLP operations, such as TF-IDF. Featuretools supports NLP operations with nlp-primitives [add-on](https://innovation.alteryx.com/natural-language-processing-featuretools/). Text variable type can be represented as strings. "
+    "## Text\n",
+    "Text is a long-form strings, that can be of any length. It is commonly used with NLP operations, such as TF-IDF. Featuretools supports NLP operations with nlp-primitives [add-on](https://innovation.alteryx.com/natural-language-processing-featuretools/)."
```
Comment (rwedge) [inline docs/source/variables.ipynb:108]: > Text is a long-form strings

maybe "long-form string"

> Featuretools supports NLP operations with nlp-primitives add-on

perhaps "with the nlp-primitives add-on"

File: docs/source/variables.ipynb (hunk 6 above)
Comment (thehomebrewnerd) [reply docs/source/variables.ipynb:9]: Not sure if it is worth including or not, but we don't currently perform variable type inference for Dask Entities. Users must specify the variable types for all the columns in the entity dataframe.

As a side note - this document will be helpful in helping users pick the right type to use.

**Comment by thehomebrewnerd** [comment]
You might also want to review the documentation in `using_dask_entitysets.rst` and update the information in the Entity Limitation sections to use `list_variable_types()` instead of the current suggestion to use `featuretools.variable_types.find_variable_types()`.

File: docs/source/variables.ipynb (hunk 2 above)
Comment (gsheni) [reply docs/source/variables.ipynb]: Fixed

File: docs/source/variables.ipynb (hunk 6 above)
Comment (gsheni) [reply docs/source/variables.ipynb]: Fixed.
I changed the wording to:

> When creating an Entity, Featuretools will attempt to infer the types of variables present. Featuretools also allows for explicitly specifying the variable types when creating the Entity."


I put a note:

>  Note: When using Dask Entities, users must explicitly specify the variable types for all columns in the Entity dataframe.

File: docs/source/variables.ipynb (hunk 7 above)
Comment (gsheni) [reply docs/source/variables.ipynb]: Fixed

**Comment by gsheni** [comment]
> You might also want to review the documentation in `using_dask_entitysets.rst` and update the information in the Entity Limitations sections to use `list_variable_types()` instead of the current suggestion to use `featuretools.variable_types.find_variable_types()`.

Fixed this.

File: docs/source/changelog.rst [hunk 8]
```diff
@@ -4,16 +4,19 @@ Changelog
...
 **Future Release**
     * Enhancements
+        * Add ``list_variable_types`` and ``graph_variable_types`` for Variable Types (:pr:`1013`)
     * Fixes
         * Improve warnings when using a Dask dataframe for cutoff times (:pr:`1026`)
     * Changes
         * Remove unnecessary ``pd.Series`` and ``pd.DatetimeIndex`` calls from primitives (:pr:`1020`, :pr:`1024`)
     * Documentation Changes
+        * Add page on Variable Types to describe some Variable Types, and util functions (:pr:`1013`)
+    * Testing Changes
```
[the remark this answers is not in the document]
Comment (gsheni) [reply docs/source/changelog.rst:14]: Removed

File: docs/source/changelog.rst [hunk 9]
```diff
@@ -4,16 +4,18 @@ Changelog
...
+        * Add ``list_variable_types`` and ``graph_variable_types`` for Variable Types (:pr:`1013`)
     * Fixes
         * Improve warnings when using a Dask dataframe for cutoff times (:pr:`1026`)
     * Changes
         * Remove unnecessary ``pd.Series`` and ``pd.DatetimeIndex`` calls from primitives (:pr:`1020`, :pr:`1024`)
     * Documentation Changes
-        * Remove featuretools enterprise from documentation (:pr:`1022`)
+        * Add page on Variable Types to describe some Variable Types, and util functions (:pr:`1013`)
     * Testing Changes
+        * Remove featuretools enterprise from documentation (:pr:`1022`)
```
Comment (rwedge) [inline docs/source/changelog.rst:15]: This is a Documentation Change

Comment (gsheni) [reply docs/source/changelog.rst:15]: Fixed.

File: featuretools/variable_types/utils.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
+import pandas as pd
+
+from featuretools.utils.gen_utils import find_descendents
+from featuretools.utils.plot_utils import (
+    check_graphviz,
+    get_graphviz_format,
+    save_graph
+)
+from featuretools.variable_types.variable import Variable
```
Comment (gsheni) [inline featuretools/variable_types/utils.py:9]: Needs to be direct import to avoid circularity

**Pull request merged**